quantmsio/operate/tools.py

```
@@ -0,0 +1,247 @@
+    mzml_directory: mzml file directory path
+    """
+    pqwriters, pqwriter_no_part, filename = init_save_info(parquet_path)
+    p = Query(parquet_path)
+    for _, table in p.iter_file(file_num=file_num):
+        refs = table["reference_file_name"].unique()
+        mzml_handlers = {ref: OpenMSHandler() for ref in refs}
+        table[["number_peaks", "mz_array", "intensity_array"]] = table[["reference_file_name", "scan"]].apply(
+            lambda x: map_spectrum_mz(
+                x["reference_file_name"],
+                x["scan"],
+                mzml_handlers,
```

Ruff

Function definition does not bind loop variable `mzml_handlers`

(B023)